spec/controllers/documents/requested_documents_controller_spec.rb

```diff
@@ -0,0 +1,71 @@
+require "rails_helper"
+
+RSpec.describe Documents::RequestedDocumentsController do
+  render_views
+
+  let(:intake) { create :intake }
+
+  before do
+    allow(subject).to receive(:user_signed_in?).and_return(true)
+    allow(subject).to receive(:current_intake).and_return intake
+  end
+
+  describe "#edit" do
+    context "with existing document uploads" do
+      it "assigns the documents to the form" do
+        doc = create :document, :with_upload, document_type: "Requested", intake: intake
+        w2_doc = create :document, :with_upload, document_type: "W-2", intake: intake
+
+        get :edit
+
+        expect(assigns(:documents)).to include(doc)
+        expect(assigns(:documents)).not_to include(w2_doc)
+      end
+    end
+
+    context "with a non-image document" do
+      let(:document_path) { Rails.root.join("spec", "fixtures", "attachments", "document_bundle.pdf") }
+
+      it "renders the thumbnails" do
+        doc = create :document, :with_upload, document_type: "Requested", intake: intake,
+          upload_path: document_path
+
+        expect { get :edit }.not_to raise_error
+        expect(response.body).to include('document_bundle.pdf')
+      end
+    end
+  end
```

Comment:
While totally fine to include it, just wanted to note that I don't feel like we need to test this for every doc upload type page. Only for a representative one.